### DISCUSSION TITLE
turbulence models

### DISCUSSION THREAD

**Discussion by cvnh** [su2code/SU2#2149] [General]
Dear SU2 developers

Just wanted to share before anyone else starts something similar. I wanted to pick up one of the development areas from #1487 to work on, and I decided to start with expanding the offer of turbulence models on SU2, specifically with 
- expanding the choice of turbulence models
- facilitate the implementation of new models, with a template-like if possible. 
I haven't found similar initiatives in my search, happy to hear if anyone else is interested to join me. 
Cheers

**Comment by bigfooted** [comment]
Hi, @cvnh ! Great that you would like to work on these topics! There was some work done to clean up the current implementation of SA and SST and to make the models consistent with literature, e.g. #1364 but you can got through the links in the issue and search the PR's and see what has been done recently.

Which turbulence models are you specifically interested in? It makes the most sense to implement models that will be used by you and the rest of the community and that augment the existing models. There are also a bunch of models that are implemented but barely visible to the user (LES, rough walls),  that would mainly be a documentation job and creating tutorial cases etc.

**Comment by cvnh** [comment]
Hi @bigfooted, thanks for the enthusiasm. Let's keep the expectations low as I'm still familiarising myself with the overall code structure and past developments on the topic. 
I'm open for suggestions from the community as personally I don't have a specific priority. I'm starting by implementing some dummy models and some tentative templates, then I'd like to implement at least one new model before revisiting the existing implementations of SA and SST, since they're surely amongst the most useful ones in practise. If the amount of work is reasonable I could implement more models, but it would be ideal to refactor some areas of the code to make it easier, but that is a bigger decision from the community. Of course significant work would be needed for validation of the implementation, but we can think about that later. 
Looking forward for your inputs!
Cheers

**Reply by pcarruscag** [reply]
The code has already been refactored to make this easier

**Reply by pcarruscag** [reply]
Join our slack channel to see the dev meeting schedules, we can point you to where you need to make changes to introduce a new model.

**Reply by cvnh** [reply]
hi Pedro, thanks, I think I figured out the basics already. My comment was about making the code more modular in general, but it is a good starting point in that the turb models are separated from the rest of the formulation. We can have a chat sometimes in the future. Cheers
